File: booksy-frontend/src/core/api/client/api-response.ts
export interface ApiResponse<T> {
  success: boolean
  data: T | null
  message?: string
  statusCode?: number
  error?: {
    code: string
    message: string
    errors?: ValidationErrors
  }
  errors?: ValidationErrors // Extracted from error.errors for easier access
  metadata?: {
    timestamp: string
    requestId: string
    path?: string
    method?: string
  }
}

export interface ErrorResponse {
  code: string
  message: string
  errors?: ValidationErrors
}

/**
 * Response metadata
 */
export interface ResponseMetadata {
  requestId: string
  timestamp: string
  duration: number
  path: string
  method: string
  version: string
}

export interface ValidationErrors {
  [field: string]: string[]
}

export interface PaginatedResponse<T> {
  items: T[]
  total: number
  page: number
  pageSize: number
  totalPages: number
  hasNextPage: boolean
  hasPreviousPage: boolean
}

export interface ApiResponseWrapper<T> extends ApiResponse<T> {
  pagination?: {
    total: number
    page: number
    pageSize: number
    totalPages: number
  }
}

File: src/Infrastructure/Booksy.API/Middleware/ApiResponse.cs

using Booksy.Core.Domain.Exceptions;

namespace Booksy.Core.Domain.Infrastructure.Middleware
{
    public class ApiResponse<TResponse> 
    {
        public bool Success { get; set; }
        public HttpStatusCode StatusCode { get; set; }
        public string Message { get; set; }
        public ApiError Error { get; set; }
        public Dictionary<string, string[]> Errors { get; set; }
        public TResponse Data { get; set; }
        public ResponseMetaData Metadata { get; internal set; }
    }
    public class ApiResponse
    {
        public bool Success { get; set; }
        public HttpStatusCode StatusCode { get; set; }
        public string Message { get; set; }
        public ApiError Error { get; set; }
        public Dictionary<string, string[]> Errors { get; set; }
        public object Data { get; set; }
        public ResponseMetaData Metadata { get; internal set; }
    }
    public class ApiError
    {
        public string Code { get; set; }
        public string Message { get; set; }
        public Dictionary<string, string[]> Errors { get; set; }

    }
}